src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/ReplicationProtectedItem/StartAzureRmRecoveryServicesAsrResynchronizeReplicationJob.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Used to initiate resync / repair replication.
+    /// </summary>
+    [Cmdlet(
+        VerbsLifecycle.Start,
+        "AzureRmRecoveryServicesAsrResynchronizeReplicationJob",
+        DefaultParameterSetName = ASRParameterSets.Default,
+        SupportsShouldProcess = true)]
+    [Alias(
+        "Start-ASRResynchronizeReplicationJob",
+        "Start-ASRResyncJob")]
+    [OutputType(typeof(ASRJob))]
+    public class StartAzureRmRecoveryServicesAsrResynchronizeReplicationJob : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets Resource Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByResourceId, Mandatory = true)]
+        public string ResourceId { get; set; }
```

Comment:
@viverm anywhere that you are using a `ResourceId` parameter, it should have the property `ValueFromPipelineByPropertyName = true` so users are able to pipe the result of `Get/Find-AzureRmResource` to a cmdlet with a `ResourceId` parameter

Reply:
Done